### ISSUE TITLE
Support mapping the same column across discriminator subclasses

### ISSUE THREAD

**Issue by parisholley** [mikro-orm/mikro-orm#2388]
**Describe the bug**
In order to make it easier for developers to understand models that rely on single-table design (in particular, managed ManyToMany tables), we "alias" some of our ManyToOne's so that you don't have to remember which logical data model maps to which generically named column. Other examples would be tables with a `to`/`from` or `parent/child` naming convention. 

```typescript
@Entity({ discriminatorValue: MemberRelationshipType.CUSTOMER })
export class BusinessCustomerMemberRelationship extends MemberRelationship<MemberRelationshipType.CUSTOMER> extends {
  @ManyToOne(() => Member, { fieldName: 'activeId', wrappedReference: true })
  customer: IdentifiedReference<Member>;

  @ManyToOne(() => Member, { fieldName: 'passiveId', wrappedReference: true })
  business: IdentifiedReference<Member>;

  // additional relationships/columns/metadata that only apply for customers
}

@Entity({ discriminatorValue: MemberRelationshipType.VENDOR })
export class BusinessVendorMemberRelationship extends MemberRelationship<MemberRelationshipType.VENDOR> extends {
  @ManyToOne(() => Member, { fieldName: 'activeId', wrappedReference: true })
  vendor: IdentifiedReference<Member>;

  @ManyToOne(() => Member, { fieldName: 'passiveId', wrappedReference: true })
  business: IdentifiedReference<Member>;

  // additional relationships/columns/metadata that only apply for vendors
}
```

When attempting to insert a discriminated subclass, the ORM will inject the activeId twice into the SQL query and it will fail. Using a getter/setter per subclass to "mimick" the alias isn't useful because it will likely cause problems when using the API (eg: filtering or populating).

**Expected behavior**
ORM should only use metadata of parent class and subclass to generate queries and access identity map.

**Versions**

```
    "@mikro-orm/cli": "^4.5.9",
    "@mikro-orm/core": "^4.5.9",
    "@mikro-orm/migrations": "^4.5.9",
    "@mikro-orm/nestjs": "^4.3.1",
    "@mikro-orm/postgresql": "^4.5.9",
    "@mikro-orm/reflection": "^4.5.9",
```

**Comment by parisholley** [comment]
Interesting thing to note, this behavior seems to be related to the fact that the class property names are different. In the example above, the `passiveId` is not inserted into the SQL twice because it is both mapped to `business`, however if you rename the fields to `business1` and `business2`, it will also duplicate `passiveId`.

**Comment by B4nan** [comment]
I believe we have a single hydration functions for this, and in general everything is processed in the same way, just the entity class is picked based on the discriminator. This is simply not supported use case right now, I believe there is even a validation that should forbid overriding properties defined in base STI entities, but here you are defining it on the same level so it fails.

**Comment by parisholley** [comment]
i'm not sure what the implications are w/r/t propagation and other internals, but as far as fixing my insertion issues, it just requires some de-dupe work inside here:

https://github.com/mikro-orm/mikro-orm/blob/f0cbcc2d977ffb8b867a21ee2ac857ef26986d73/packages/knex/src/AbstractSqlDriver.ts#L243-L274

i have a local patch working that i'll run with and see if i run into any weird behavior before considering a PR

**Comment by B4nan** [comment]
Fixed in v6 via #4769
